iris_pipeline/merge_subarrays/merge_subarrays.py

```diff
@@ -0,0 +1,43 @@
+import numpy as np
+
+from jwst.stpipe import Step
+from jwst import datamodels
+
+__all__ = ["MergeSubarraysStep"]
+
+
+class MergeSubarraysStep(Step):
+    """
+    ParseSubarrayMapStep: Parse a subarray map
+    extension, if available, and create header metadata
+    and data quality flag accordingly
+    """
+
+    def process(self, input):
+
+        input = datamodels.open(input)
+
+        # If single input, wrap in a ModelContainer
+        if not isinstance(input, datamodels.ModelContainer):
+            self.log.info("No subarray files provided, return the original model")
+            return input
+        else:
+            input_models = input
+
+        result = input_models[0].copy()
+
+        # Assume subarrays are in order
+        for i_sub in range(1, len(input_models)):
```

Comment:
@arunsurya77 what is the keyword in the subarrays headers that specifies the subarray number? I.e. subarray 1 should have a keyword which say subarray_num=1 or something similar.

Reply:
This could be done. We already tag the number in the filename and the mask extension. So if required we can put it in the header too just for some redundancy.

Reply:
there is no mask extension in the subarray file. I think we should not rely on filenames for important metadata. We need a metadata keyword in the header. I think we should decide a name for this keyword (`SUBARRID`)? I can implement it into `IRISImageModel`, but to which document we can add this?

Reply:
ok, I implemented this, see dee446070dcec82f2ffe59fe75a409a9eed0bbcc